helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java

```diff
@@ -176,8 +176,8 @@ public void testP2PStateTransitionEnabled() {
 
     // The success rate really depends on how quick participant act in relationship with controller.
     // For now, we set 90% threshold.
```

Comment:
Congrats on first code-review. Can you please change the comment as well?  also i don't have full context on why 80 would work and not 90.

Reply:
Thanks for the quick review! I'm suspecting that this unit test is buggy due to the limited github action resource. As the previous author mentioned, this would cause the state transition of participant falls behind the time when the `verifyP2PEnabled()` method reads the ZK. Thus, lowering down the threshold could potentially avoid this issue. To monitor the effect of this change, another line is added to log the number of successful P2P transition and total transitions expected, which would make debugging this issue a little bit easier if it comes back in the future.

Reply:
Just found out that this is not the root cause. Rolled out another commit to resolve the race condition instead of just lowering the threshold.